velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -929,6 +930,68 @@ PlanBuilder& PlanBuilder::localMerge(const std::vector<std::string>& keys) {
   return *this;
 }
 
+PlanBuilder& PlanBuilder::expand(
+    const std::vector<std::vector<std::string>>& projections,
+    TypePtr groupIdColumnType) {
+  VELOX_CHECK(!projections.empty(), "projections must not be empty.");
+  const auto numColumns = projections[0].size();
+  const auto numRows = projections.size();
+  std::vector<std::string> aliasNames;
+  aliasNames.reserve(numColumns);
+  std::vector<std::string> originalNames;
+  originalNames.reserve(numColumns);
+  std::vector<TypePtr> types;
+  types.reserve(numColumns);
+  const auto& inputRowType = planNode_->outputType();
+
+  std::vector<std::vector<core::TypedExprPtr>> projectExprs;
+  projectExprs.reserve(projections.size());
+
+  for (auto i = 0; i < numRows; i++) {
+    std::vector<core::TypedExprPtr> projectExpr;
+    for (auto j = 0; j < numColumns; j++) {
+      auto untypedExpression = parse::parseExpr(projections[i][j], options_);
+      auto typedExpression = inferTypes(untypedExpression);
+
+      if (i == 0) {
+        if (auto fieldExpr = dynamic_cast<const core::FieldAccessExpr*>(
+                untypedExpression.get())) {
+          if (fieldExpr->alias().has_value()) {
+            aliasNames.push_back(fieldExpr->alias().value());
+          } else {
+            aliasNames.push_back(fieldExpr->getFieldName());
+          }
+          originalNames.push_back(fieldExpr->getFieldName());
+        } else if (untypedExpression->alias().has_value()) {
+          aliasNames.push_back(untypedExpression->alias().value());
+          originalNames.push_back(untypedExpression->alias().value());
```

Comment:
@JkSelf I'm confused. Why do we add alias to originalNames ?

Also, what happens if untypedExpression doesn't have an alias? If this is not allowed, then perhaps, add a check.

Reply:
@mbasmanova https://github.com/facebookincubator/velox/pull/5403/files#diff-fd8dcf1389140cd0b653207240aa59d35e7959de82ed4e08369b78a85737f04fR976 To obtain the type based on the column name, we need to retrieve the original names since the inputRowType does not include alias names. Therefore, it is necessary to store the original names for reference.

Reply:
> Therefore, it is necessary to store the original names for reference.

This could be so, but the code doesn't store the original names. It stores the alias in the variable named originalNames. This is confusing, no? CC: @rui-mo 

> originalNames.push_back(untypedExpression->alias().value());